### ISSUE TITLE
Properties which implment KeyedCollection should emit uniqueItems: true

### ISSUE THREAD

**Issue by freeranger** [domaindrivendev/Swashbuckle.AspNetCore#945]
Currently Properties who's type are or implement KeyedCollection are emitted as arrays without uniqueItems: true this results in API documentation that does not properly represent the behavior of the type.

This is similar to the problem raised in #430 for properties implementing ISet

**Comment by freeranger** [comment]
In the mean time, here is a filter which will do the job:
```
    public class KeyedCollectionSchemaFilter : ISchemaFilter
    {
        /// <inheritdoc />
        public void Apply(Schema model, SchemaFilterContext context)
        {
            if (context.SystemType.IsGenericType && context.SystemType.GetGenericTypeDefinition() == typeof(KeyedCollection<,>) || IsSubclassOfGeneric(context.SystemType, typeof(KeyedCollection<,>)))
            {
                model.UniqueItems = true;
            }
        }

        private static bool IsSubclassOfGeneric(Type type, Type generic)
        {
            while (type != null && type != typeof(object))
            {
                var typeToCheck = type.IsGenericType ? type.GetGenericTypeDefinition() : type;
                if (typeToCheck == generic)
                {
                    return true;
                }
                type = type.BaseType;
            }
            return false;
        }
    }
```

**Comment by domaindrivendev** [comment]
Fixed by #1948
